src/api/app/jobs/send_event_emails.rb

```diff
@@ -1,24 +1,24 @@
+# TODO: Rename this to something more appropriate like "ProcessEvents"
 class SendEventEmails
   # we don't need this outside of the migration - but we need to be able
   # to load old jobs from the database (and mark their events) before deleting
   # (see 20151030130011_mark_events)
   attr_accessor :event
 
   def perform
-    Event::Base.where(mails_sent: false).order(:created_at).limit(1000).lock(true).each do |event|
-      event.mails_sent = true
-      begin
-        event.save!
-      rescue ActiveRecord::StaleObjectError
-        # if someone else saved it too, better don't continue - we're not alone
-        return false
-      end
+    events_relation = Event::Base.where(mails_sent: false).order(:created_at).limit(1000)
+    event_ids = events_relation.pluck(:id)
+    events_relation.update_all(mails_sent: true)
+
+    Event::Base.find(event_ids).each do |event|
       subscribers = event.subscribers
       next if subscribers.empty?
+
       EventMailer.event(subscribers, event).deliver_now
+
       create_rss_notifications(event)
+      event.create_project_log_entry if event.needs_logging?
```

Comment:
if the **needs_logging?** call is inside the **create_project_log_entry**, then you don't need it here...